src/rrt-viewer/RRTWidget.cpp

```diff
@@ -60,6 +63,45 @@ void RRTWidget::reset() {
     update();
 }
 
+void RRTWidget::saveObstacles() {
+    ofstream file;
+    file.open("rrtconfig.txt");
+    ObstacleGrid& grid = _stateSpace->obstacleGrid();
+    cout << grid.height() << " " << grid.width() << endl;
+    for (int j = 0; j < grid.discretizedHeight(); j++) {
+        for (int i = 0; i < grid.discretizedWidth(); i++) {
+            if (grid.obstacleAt(i, j)) {
+                file << "1";
+            } else {
+                file << "0";
+            }
+        }
+    }
+    file.close();
+    cout << "Obstacles saved" << endl;
+}
+
+void RRTWidget::loadObstacles() {
+    string line;
+    ifstream file("rrtconfig.txt");
+    ObstacleGrid& grid = _stateSpace->obstacleGrid();
+    char x;
+    if (file.is_open()) {
+        for (int j = 0; j < grid.discretizedHeight(); j++) {
+            for (int i = 0; i < grid.discretizedWidth(); i++) {
+                file >> x;
+                if ('1' == x) {
+                    grid.obstacleAt(i, j) = true;
+                } else if ('0' == x) {
```

Comment:
Can you use double quotes everywhere or single quotes everywhere :P

Reply:
True